libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -1383,7 +1382,7 @@ MQTTStatus_t MQTT_GetConnectPacketSize( const MQTTConnectInfo_t * pConnectInfo,
         LogError( ( "The Will Message length must not exceed %d. "
                     "pWillInfo->payloadLength=%lu.",
                     UINT16_MAX,
```

Comment:
Should the UINT16_MAX value be logged as `%u`?
```suggestion
        LogError( ( "The Will Message length must not exceed %u. "
                    "pWillInfo->payloadLength=%lu.",
                    UINT16_MAX,
```

Reply:
In gcc's `stdint.h` I see this:
```
#define UINT8_MAX         255
#define UINT16_MAX        65535
#define UINT32_MAX        4294967295U
#define UINT64_MAX        18446744073709551615ULL
```
`UINT16_MAX` doesn't seem to be declared unsigned as `UINT32_MAX` is, so I think it's best to leave as is.